WooCommerce/src/main/kotlin/com/woocommerce/android/ui/bookings/filter/productname/BookingServiceEventFilterUiState.kt

```diff
@@ -0,0 +1,59 @@
+package com.woocommerce.android.ui.bookings.filter.productname
+
+import com.woocommerce.android.R
+import com.woocommerce.android.model.UiString.UiStringRes
+import com.woocommerce.android.model.UiString.UiStringText
+import com.woocommerce.android.ui.bookings.filter.BookingFilterListItem
+import org.wordpress.android.fluxc.network.rest.wpcom.wc.bookings.BookingsFilterOption
+
+data class BookingServiceEventFilterUiState(
+    val availableProducts: List<BookableProduct> = emptyList(),
+    val selectedProducts: BookingsFilterOption.ServiceEvents = BookingsFilterOption.ServiceEvents.DEFAULT,
+    val searchQuery: String = "",
+    val onProductSelected: (BookableProduct?) -> Unit = {},
+    val onSearchQueryChanged: (String) -> Unit = {},
+) {
+    val items: List<BookingFilterListItem> = filteredProducts().map { product ->
+        BookingFilterListItem(
+            title = product.name?.let { UiStringText(it) }
+                ?: UiStringRes(R.string.bookings_filter_default),
+            selected = isSelected(product),
+            onClick = { onProductSelected(product) }
+        )
+    }
+
+    private fun filteredProducts(): List<BookableProduct> {
+        val selectedProductsList = selectedProducts.values.map {
+            BookableProduct(id = it.productId, name = it.productName)
+        }
+
+        val productsToFilter = (
+            listOf(BookableProduct.Any) +
+                selectedProductsList.filter { it !in availableProducts } +
+                availableProducts
+            ).distinctBy { it.id }
+
+        return if (searchQuery.isBlank()) {
+            productsToFilter
+        } else {
+            productsToFilter.filter { product ->
+                product.name?.contains(searchQuery, ignoreCase = true) ?: false
```

Comment:
```suggestion
                product.name?.contains(searchQuery.trim(), ignoreCase = true) ?: false
```

I would suggest trimming the query if we are doing local search only.